### ISSUE TITLE
Unable to create chain with StreamProxy (duplexify)

### ISSUE THREAD

**Issue by rhodgkins** [uhop/stream-chain#4]
Hi,

I'm trying to use this library in relation to a streamed Google gRPC response that uses [`StreamProxy`](https://github.com/googleapis/gax-nodejs/blob/bee3781ee4741a845597dad79bf4a0998c716fc1/src/streamingCalls/streaming.ts#L88) (and [`duplexify`](https://github.com/mafintosh/duplexify)), but there the below check when creating a chain that prevents this... 

https://github.com/uhop/stream-chain/blob/49b851fa38c0a53f6c62fa3859a0bfbb4f7008be/index.js#L97-L105

Any ideas on a way around this? (Currently wrapping a pointless `PassThrough` with the `StreamProxy` in a `NodeJS.pipeline` to get around it but this isn't ideal obviously!)

**Comment by uhop** [comment]
The check in question is to make sure that arguments are functions, arrays, or streams. As far as I can tell `StreamProxy` (and `duplexify`) are objects, which do not implement stream protocols and are not based on any standard Node stream.

**Comment by rhodgkins** [comment]
`duplexity` uses [`readable-stream`](https://www.npmjs.com/package/readable-stream) which is a package mirroring the standard Node streams implementing all the stream protocols based on the standard Node stream.

**Comment by rhodgkins** [comment]
I'd disagree its third party as its ported from NodeJS, anyway what about matching NodeJS checks for streams?

https://github.com/nodejs/node/blob/a82c1a1da710e83bebd4431a0220e45562a310a3/lib/internal/streams/utils.js#L14-L44

There's no need for the `instanceof Transform` if its a `Duplex` stream, so check could just become:

```js
if ( 
   isDuplexNodeStream(fn) || 
   (!index && isReadableNodeStream(fn)) || 
   (index === fns.length - 1 && isWritableNodeStream(fn)) 
 ) { 
   return fn; 
 } 
 throw Error('Arguments should be functions, arrays or streams.'); 
```

**Comment by uhop** [comment]
Sounds like a plan. I'll add it to the next release.

**Comment by uhop** [comment]
Released as 2.2.5.
